File: csharp/DiagramPrinter/DiagramSummary.cs
namespace DiagramPrinter;

public class DiagramSummary
{
    private readonly string _language;
    private string contents = "";

    public DiagramSummary(string language)
    {
        _language = language;
    }

    public void AddHeader(string summaryInformation)
    {
        contents += summaryInformation;
        contents += "\n";
    }

    public void AddTitle(string name, string serialNumber)
    {
        contents += name;
        contents += "\n";
        contents += serialNumber;
        contents += "\n";
    }

    public void AddImage(PngDocument flowchart)
    {
        contents += flowchart.Filename();
    }

    public string Export()
    {
        // imagine a lot more detail and complexity here
        return contents;
    }
}

File: cpp/src/diagram_summary.cpp
#include "diagram_summary.h"

DiagramSummary::DiagramSummary(const std::string& language)
    : _language(language)
    , contents("")
{
}

void DiagramSummary::AddHeader(const std::string& summaryInformation)
{
    contents += summaryInformation;
    contents += "\n";
}

void DiagramSummary::AddTitle(const std::string& name, const std::string& serialNumber)
{
    contents += name;
    contents += "\n";
    contents += serialNumber;
    contents += "\n";
}

void DiagramSummary::AddImage(const PngDocument *flowchart)
{
    contents += flowchart->Filename();
}

std::string DiagramSummary::Export()
{
    // imagine a lot more detail and complexity here
    return contents;
}
